src/db.rs

```diff
@@ -125,6 +126,46 @@ impl Database {
             });
         })
     }
+
+    fn checkpoint(
```

Comment:
it seems the implementation of the `checkpoint` in `db.rs` and `state_db.rs` are totally same, so we can use it like other functions in my PR #58, like `close` function implement it in `common_db.rs` and use it.

Reply:
Thank you for your suggestion. Done.